microcloud/mdns/lookup.go

```diff
@@ -148,7 +148,26 @@ func Lookup(ctx context.Context, iface *net.Interface, service string, size int)
 	params.Interface = iface
 	params.Entries = entriesCh
 	params.Timeout = 100 * time.Millisecond
-	err := mdns.Query(params)
+	ipv4Supported, ipv6Supported, err := checkIPStatus(iface.Name)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to check IPv6 status: %w", err)
```

Comment:
Should probably have `return nil, fmt.Errorf("Failed to check IP status: %w", err)"` as the caller doesn't no the error cause here.

Reply:
fixed